pandas/core/arrays/timedeltas.py

```diff
@@ -288,6 +317,21 @@ def _evaluate_with_timedelta_like(self, other, op):
 
         return NotImplemented
 
+    __mul__ = _wrap_tdi_op(operator.mul)
```

Comment:
we should use the already existing machinery and simply implement this ``_create_arithmetic_method``
and a limited form of:    

```
@classmethod
    def _add_arithmetic_ops(cls):
```

otherwise you have different styles here than elsewhere.

Reply:
We use this pattern in plenty of places: nattype, Timedelta, field accessors TDA/DTA/PA, ... Using the classmethod in this context doesn't add anything except a layer of indirection.

Reply:
> as this IS using the mixin that implements this

No, it isn't.  It inherits from ExtensionOpsMixin to use _add_comparison_comps, NOT _add_arithmetic_ops.  In fact if were to try to use _add_arithmetic_ops that would be working at cross-purposes with `_add_datetimelike_methods`, which specifically implements add/sub-like methods.

I'm as big a fan of internal consistency and de-duplication as the next guy, but this isn't the place to make that stand.

Reply:
> I think the goal is to rework this a bit at some point, to not wrap TimedeltaIndex here, rather the other way around

Correct